### DISCUSSION TITLE
Blade Componentes Doesn't Work Only Production.

### DISCUSSION THREAD

**Discussion by devajmeireles** [laravel/framework#38601] [Q&A]
This case is really intriguing...

Hello community!

I'm a PHP developer and I don't usually ask for much help because from the knowledge I have, I always manage to do well reading the tools' documentation.

I recently migrated to Laravel and in order to take advantage of how good the software I'm building is, I decided to build the system on a VPS and simulate production use.

For that I decided to use Docker, so that the same environment I had in development, I could experiment in production, but I came across a problem which I can't understand why it happens, or how it happens. I will explain in detail:

The system menu of this system was through a component, and to better organize I created a namespace (folder) inside: `App/View/Components/`, looking like this: `App/Views/Components/Category/CategoryMenu.php`

This `CategoryMenu.php` file has the following structure:
![image](https://user-images.githubusercontent.com/60591772/131421352-da296297-e117-4786-85fb-27ef7bf04627.png)

... **And I invoke this component in the file**: `/resources/views/layouts.blade.php`, looking like this:
`<x-category.categorymenu order="id" direction="asc" limit="6" />`

... I also have tried like:
<x-category.categorymenu order="id" direction="asc" limit="6"></x-category.categorymenu>

The component's HTML structure is simple, it's just an organized foreach, so:
![image](https://user-images.githubusercontent.com/60591772/131421524-a3a0e93c-2e83-4586-b2bb-b1bc4e2dcc36.png)

### The big question:

Why this same component doesn't give any kind of problem in development environment? Considering that the use of the **development environment is done in Docker, which in this way replicates THE SAME ENVIRONMENT for the structure in production?**

### HTML Renderization:

Route access: `/registro` **(DEVELOPMENT):**
![image](https://user-images.githubusercontent.com/60591772/131421783-e37a02fd-3325-42a1-aec4-97214ca36437.png)

**WORK'S FINE!**

Route access: `/registro` **(PRODUCTION):**
![image](https://user-images.githubusercontent.com/60591772/131421839-db61cfb1-cb0f-472a-84eb-a6e378f9d03c.png)

### Observations:

1. Any attempt to code "dd" within:
`App/Views/Components/Category/CategoryMenu.php` is ignored and does not kill the code.

2. Laravel creates cache files with user:group:
nobody:nogroup

---

Sorry this post was long, I wanted to explain in detail.

Does anyone have an idea what this problem might be?

**Comment by dennisprudlo** [comment]
Have you tried to invoke the `$categories` variable in the components blade view since it's a public function in the class? (https://laravel.com/docs/8.x/blade#component-methods)
